05_testing_and_ci/README.md

```diff
@@ -0,0 +1,13 @@
+# Testing and CI
+
+Learning goals:
+
+- Understanding why and how to write tests
+- Understanding the different types of tests and their relevance in testing scientific software
+- Becoming familiar with Python testing frameworks `pytest` and `unittest`
```

Comment:
Let's try to iterate these a bit. I know, they are neither perfect in other chapters (we will need to tune next year).
- "Understanding why and how to write tests" ... is not concise enough, especially the "how".
- "... the different types of tests and their relevance ..." ... sounds like we would really touch all kinds of tests. We don't talk about system tests, performance tests, ... Why not making things explicit here? "Scientific software" -> either research software or simulation software. I prefer the latter. Maybe at some point, we add simulation specific things here, like how to test floating point things, ill-conditioned problems, ...
- "Becoming familiar" is also not very concise. What exactly do students learn?

Reply:
Second iteration:
* Understanding the relevance of testing simulation software
* Understanding the concepts of unit testing, integration testing and regression testing
* How to use the Python testing frameworks `pytest` and `unittest` to generate a automated testing workflow
Comments?